### PR TITLE
Additional Tests

### PR DESCRIPTION
**Pull request by gannan08**

### CODE REVIEW

File: tests/30-main.spec.js [hunk 1]
```diff
@@ -254,4 +255,450 @@ describe('main', () => {
...
+    err.should.be.instanceof(TypeError);
+    err.message.should.contain('"credential" must be an object');
+  });
+
+  it('should fail to verify if credential is not an object for ' +
+    '"assertRevocationList2020Context"', async () => {
+    let err;
+    let result;
+    try {
+      result = assertRevocationList2020Context('{credential}');
```
Comment (dlongley) [inline tests/30-main.spec.js]: There are still issues like this one that were flagged in the last PR that haven't been addressed.

Comment (gannan08) [reply tests/30-main.spec.js]: @LoveishSarolia Please fix outstanding issues.

Comment (LoveishSarolia) [reply tests/30-main.spec.js]: I believe I fixed them a while ago and pushed them to the #add-tests-pr branch. If there are any other issues, I am not seeing them. Am I correct in thinking these are the only issues that needed to be addressed?

File: tests/30-main.spec.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ import {
 import {extendContextLoader} from 'jsonld-signatures';
 import {constants, contexts} from 'vc-revocation-list-context';
 import vc from 'vc-js';
+import {assertRevocationList2020Context, getCredentialStatus} from '../main';
```
Comment (gannan08) [inline tests/30-main.spec.js]: Missing file extension.

File: tests/30-main.spec.js [hunk 3]
```diff
@@ -254,4 +255,450 @@ describe('main', () => {
...
+    let err;
+    let result;
+    try {
+      delete credential.credentialSubject[0];
+      result = await checkStatus({
+        credential, documentLoader, verifyRevocationListCredential: false});
+    } catch(e) {
+      err = e;
+    }
+    console.log(result);
```
Comment (gannan08) [inline tests/30-main.spec.js]: Remove logging.

File: tests/30-main.spec.js [hunk 4]
```diff
@@ -254,4 +255,450 @@ describe('main', () => {
...
+    let err;
+    let result;
+    try {
+      delete credential.type[1];
+      result = await checkStatus({
+        credential, documentLoader, verifyRevocationListCredential: false});
+    } catch(e) {
+      err = e;
+    }
+    console.log(result);
```
Comment (gannan08) [inline tests/30-main.spec.js]: Remove logging.

**Review by gannan08 — COMMENTED** [review]
@LoveishSarolia Please fix issues.

File: tests/30-main.spec.js (hunk 2 above)
Comment (LoveishSarolia) [reply tests/30-main.spec.js]: Fixed.

File: tests/30-main.spec.js (hunk 3 above)
Comment (LoveishSarolia) [reply tests/30-main.spec.js]: Fixed.

File: tests/30-main.spec.js (hunk 4 above)
Comment (LoveishSarolia) [reply tests/30-main.spec.js]: Fixed.

File: tests/30-main.spec.js [hunk 5]
```diff
@@ -254,4 +255,448 @@ describe('main', () => {
...
+    } catch(e) {
+      err = e;
+    }
+    should.exist(err);
+    should.not.exist(result);
+    err.should.be.instanceof(TypeError);
+    err.message.should.contain('"@context" must be an array');
+  });
+
+  it('should fail to test the first "@context" value correctly in ' +
```
Comment (dlongley) [inline tests/30-main.spec.js]:
```suggestion
  it('should fail when the first "@context" value is unexpected in ' +
```

File: tests/30-main.spec.js [hunk 6]
```diff
@@ -254,4 +255,448 @@ describe('main', () => {
...
+    } catch(e) {
+      err = e;
+    }
+    should.exist(err);
+    should.not.exist(result);
+    err.should.be.instanceof(TypeError);
+    err.message.should.contain('"@context" must be an array');
+  });
+
+  it('should fail to test the first "@context" value correctly in' +
```
Comment (dlongley) [inline tests/30-main.spec.js]:
```suggestion
  it('should fail when the first "@context" value is unexpected in' +
```

File: tests/30-main.spec.js [hunk 7]
```diff
@@ -254,4 +255,448 @@ describe('main', () => {
...
+    } catch(e) {
+      err = e;
+    }
+    should.exist(err);
+    should.not.exist(result);
+    err.should.be.instanceof(Error);
+    err.message.should.contain('first "@context" value');
+  });
+
+  it('should fail to validate if "credentialStatus" does not ' +
```
Comment (dlongley) [inline tests/30-main.spec.js]:
```suggestion
  it('should fail when "credentialStatus" does not ' +
```

Comment (dlongley) [reply tests/30-main.spec.js]: I noticed "validate" and "verify" being used interchangeably here in the test names for helper functions -- so perhaps best to just remove them and talk about failures.

File: tests/30-main.spec.js [hunk 8]
```diff
@@ -254,4 +255,448 @@ describe('main', () => {
...
+      delete credential.credentialStatus;
+      result = statusTypeMatches({credential});
+    } catch(e) {
+      err = e;
+    }
+    should.not.exist(err);
+    result.should.equal(false);
+  });
+
+  it('should fail to verify if "credentialStatus" is not an object in ' +
```
Comment (dlongley) [inline tests/30-main.spec.js]:
```suggestion
  it('should fail when "credentialStatus" is not an object in ' +
```

File: tests/30-main.spec.js [hunk 9]
```diff
@@ -254,4 +255,448 @@ describe('main', () => {
...
+    } catch(e) {
+      err = e;
+    }
+    should.exist(err);
+    should.not.exist(result);
+    err.should.be.instanceof(Error);
+    err.message.should.contain('"credentialStatus" is invalid');
+  });
+
+  it('should return false if "CONTEXTS.RL_V1" is not in ' +
```
Comment (dlongley) [inline tests/30-main.spec.js]:
```suggestion
  it('should return false when "CONTEXTS.RL_V1" is not in ' +
```

File: tests/30-main.spec.js [hunk 10]
```diff
@@ -254,4 +255,448 @@ describe('main', () => {
...
+      };
+      result = statusTypeMatches({credential});
+    } catch(e) {
+      err = e;
+    }
+    should.not.exist(err);
+    result.should.equal(false);
+  });
+
+  it('should fail to validate if "CONTEXTS.RL_V1" is not in "@contexts" ' +
```
Comment (dlongley) [inline tests/30-main.spec.js]:
```suggestion
  it('should fail when "CONTEXTS.RL_V1" is not in "@contexts" ' +
```

File: tests/30-main.spec.js [hunk 11]
```diff
@@ -254,4 +255,448 @@ describe('main', () => {
...
+    } catch(e) {
+      err = e;
+    }
+    should.exist(err);
+    should.not.exist(result);
+    err.should.be.instanceof(TypeError);
+    err.message.should.contain('@context" must include');
+  });
+
+  it('should fail to verify if credentialStatus is not an object for ' +
```
Comment (dlongley) [inline tests/30-main.spec.js]:
```suggestion
  it('should fail when credentialStatus is not an object for ' +
```

File: tests/30-main.spec.js [hunk 12]
```diff
@@ -254,4 +255,448 @@ describe('main', () => {
...
+    } catch(e) {
+      err = e;
+    }
+    should.exist(err);
+    should.not.exist(result);
+    err.should.be.instanceof(Error);
+    err.message.should.contain('"credentialStatus" is missing or invalid');
+  });
+
+  it('should fail to validate if documentLoader is not a function',
```
Comment (dlongley) [inline tests/30-main.spec.js]:
```suggestion
  it('should fail to verify when documentLoader is not a function',
```

File: tests/30-main.spec.js [hunk 13]
```diff
@@ -254,4 +255,448 @@ describe('main', () => {
...
+      result.should.have.property('verified');
+      result.verified.should.be.a('boolean');
+      result.verified.should.be.false;
+      result.should.have.property('error');
+      result.error.should.be.instanceof(TypeError);
+      result.error.message.should.contain(
+        '"documentLoader" must be a function');
+    });
+
+  it('should fail to verify if suite is not an object or array of ' +
```
Comment (dlongley) [inline tests/30-main.spec.js]:
```suggestion
  it('should fail to verify when suite is not an object or array of ' +
```

File: tests/30-main.spec.js [hunk 14]
```diff
@@ -254,4 +255,448 @@ describe('main', () => {
...
+    result.should.have.property('verified');
+    result.verified.should.be.a('boolean');
+    result.verified.should.be.false;
+    result.should.have.property('error');
+    result.error.should.be.instanceof(TypeError);
+    result.error.message.should.contain(
+      '"suite" must be an object or an array of objects');
+  });
+
+  // FIXME: Need to invoke the error message alone with the verification
```
Comment (dlongley) [inline tests/30-main.spec.js]:
```suggestion
  // FIXME: Need to invoke the error message alone with the verification
```

What does it mean to invoke an error message? Can this FIXME language be more clear? Or, better yet, can we get it addressed in this PR?

File: tests/30-main.spec.js [hunk 15]
```diff
@@ -254,4 +255,448 @@ describe('main', () => {
...
+  it.skip('should fail to verify if "RevocationList2020Credential" ' +
```
Comment (dlongley) [inline tests/30-main.spec.js]:
```suggestion
  it.skip('should fail to verify when "RevocationList2020Credential" ' +
```

File: tests/30-main.spec.js [hunk 16]
```diff
@@ -254,4 +255,448 @@ describe('main', () => {
...
+    result.should.have.property('verified');
+    result.verified.should.be.a('boolean');
+    result.verified.should.be.false;
+    result.should.have.property('error');
+    result.error.should.be.instanceof(Error);
+    result.error.message.should.contain(
+      '"RevocationList2020Credential" not verified');
+  });
+
+  // FIXME: Need to invoke the error message alone with the verification
```
Comment (dlongley) [inline tests/30-main.spec.js]:
```suggestion
  // FIXME: Need to invoke the error message alone with the verification
```

Same issue as above.

File: tests/30-main.spec.js [hunk 17]
```diff
@@ -254,4 +255,448 @@ describe('main', () => {
...
+  it.skip('should fail to verify if "RevocationList2020" ' +
```
Comment (dlongley) [inline tests/30-main.spec.js]:
```suggestion
  it.skip('should fail to verify when "RevocationList2020" ' +
```

**Review by dlongley — CHANGES_REQUESTED** [review]
Looking good! Just a few naming nits to clean up and then those couple of FIXMEs need to get addressed.

File: tests/30-main.spec.js (hunk 16 above)
Comment (LoveishSarolia) [reply tests/30-main.spec.js]: This test was deleted because it was testing the wrong thing.

**Review by dlongley — APPROVED** [review]
Thanks, @LoveishSarolia!

**Comment by dlongley** [comment]
This is ready for merge provided that it doesn't cause any trouble with the other outstanding PRs.

**Review by mattcollier — APPROVED** [review]

**Pull request merged**